llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -14967,6 +14967,9 @@ bool SCEVWrapPredicate::implies(const SCEVPredicate *N,
   if (Start->getType()->isPointerTy() != OpStart->getType()->isPointerTy())
     return false;
 
+  if (Start->getType()->isPointerTy() && Start->getType() != OpStart->getType())
```

Comment:
nit: might be worth adding a comment here to say that this rejects pointers in different address spaces?